### PR TITLE
Update unit tests for API plugin mixin class

### PR DESCRIPTION
**Pull request by SchrodingersGat**
- API at previous target URL has changed
- Simplier to use the github API as a test case

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/3328"></a>

### CODE REVIEW

File: InvenTree/plugin/base/integration/test_mixins.py [hunk 1]
```diff
@@ -232,30 +242,32 @@ def test_api_call(self):
...
+        result = self.mixin.api_call('https://api.github.com/orgs/inventree', endpoint_is_url=True)
         self.assertTrue(result)
 
         # api_call with post and data
         result = self.mixin.api_call(
-            'api/users/',
-            data={"name": "morpheus", "job": "leader"},
-            method='POST'
+            'repos/inventree/InvenTree',
+            method='GET'
```
Comment (matmair) [inline InvenTree/plugin/base/integration/test_mixins.py:261]: @SchrodingersGat this test is not really doing anything valuable anymore as the previous implementation did a POST request to ensure that works too. Could we change that back to ensure all methods work?

**Review by matmair — COMMENTED** [review]
@SchrodingersGat was afk for a while - I think these changes weakened coverage meaning for the tests.

File: InvenTree/plugin/base/integration/test_mixins.py (hunk 1 above)
Comment (SchrodingersGat) [reply InvenTree/plugin/base/integration/test_mixins.py:261]: Can do, I was needing the tests to pass and waiting for your return.

Comment (matmair) [reply InvenTree/plugin/base/integration/test_mixins.py:261]: Should be resolved in https://github.com/inventree/InvenTree/pull/3328

**Pull request merged**